### PR TITLE
fix: do not modify custom type names

### PR DESCRIPTION
**Pull request by digorithm**
# Problem

Suppose the following two structs in Sway:

```Rust
pub struct Campaign {
    author: Identity,
    asset: ContractId,
    beneficiary: Identity,
    claimed: bool,
    deadline: u64,
    state: u64, 
    target_amount: u64,
    total_pledge: u64,
}

pub struct Campaigns {
    active: [u64; 1],
    completed: [u64; 1],
}
```

Given the current state of the SDK, this would lead to this error:

image

That's because `Campaigns` was being turned into `Campaign` due to `Inflector`'s `.to_class_case()`, and then conflicting with the original `Campaign`. 

Fiddling with the casing chosen by the Sway user is bad and not wanted, whatever the Sway user decides (and the _compiler_ allows) should be what is generated by the SDK, otherwise, there will be a discrepancy between what the user chose and what the SDK is generating, which isn't good.

# Solution

This PR simply removes the mutation of casing altogether. No more `.to_class_case()`. A very few tests were updated to use proper casing and not to expect casing mutation at all. The other tests don't break with the removal of `.to_cass_case()` and cases where the user defines structs such as `Campaign` and `Campaigns` are expected to work.

### CODE REVIEW

File: packages/fuels-core/src/code_gen/functions_gen.rs [hunk 1]
```diff
@@ -741,32 +734,6 @@ pub fn hello_world(
...
-            name: String::from("unused"),
-            type_field: String::from("enum babies"),
-            components: None,
-        };
-        let enum_name = Some(&enum_prop);
-        let result = expand_input_param(&def, "unused", &enum_type, &enum_name);
-        // Notice the removed plural!
-        assert_eq!(result.unwrap().to_string(), "Baby");
-    }
-    #[test]
```
Comment (matt-user) [inline packages/fuels-core/src/code_gen/functions_gen.rs:769]: Why did you remove the #[test] (macro?) from test_expand_input_param_struct_wrong_name()?

File: packages/fuels-core/src/code_gen/custom_types_gen.rs [hunk 2]
```diff
@@ -574,11 +568,11 @@ mod tests {
     fn test_expand_custom_struct_with_struct() {
         let p = Property {
             name: String::from("unused"),
-            type_field: String::from("struct cocktail"),
+            type_field: String::from("struct Cocktail"),
```
Comment (matt-user) [inline packages/fuels-core/src/code_gen/custom_types_gen.rs:571]: Did you rename all of the type fields to match the expected ClassCase types for the tests, now that there is no longer a conversion to ClassCase?

File: packages/fuels-abigen-macro/tests/test_projects/enum_as_input/src/main.sw [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+    Three: bool,
 }
 
 enum UnitEnum {
-    one: (),
-    two: (),
-    three: (),
+    One: (),
+    Two: (),
+    Three: (),
```
Comment (matt-user) [inline packages/fuels-abigen-macro/tests/test_projects/enum_as_input/src/main.sw:12]: Why rename these enum types?  Is it because the tests expect them to be in ClassCase?

File: packages/fuels-core/src/code_gen/functions_gen.rs [hunk 4]
```diff
@@ -741,32 +734,6 @@ pub fn hello_world(
         assert_eq!(result.unwrap().to_string(), ":: std :: vec :: Vec < u64 >");
     }
     #[test]
-    fn test_expand_input_param_custom_type() {
```
Comment (matt-user) [inline packages/fuels-core/src/code_gen/functions_gen.rs:744]: Why delete the test case?  Couldn't you rename the variable types as you did in other places?

**Review by matt-user — COMMENTED** [review]
Not so much a review... just curious about the code.

File: packages/fuels-core/src/code_gen/functions_gen.rs (hunk 1 above)
Comment (digorithm) [reply packages/fuels-core/src/code_gen/functions_gen.rs:769]: Uh... weird diff. It's still there, see whole file: https://github.com/FuelLabs/fuels-rs/blob/e07c117d2de67f66a1e67c3959b0d1834b1e6f3f/packages/fuels-core/src/code_gen/functions_gen.rs#L736

File: packages/fuels-abigen-macro/tests/test_projects/enum_as_input/src/main.sw (hunk 3 above)
Comment (digorithm) [reply packages/fuels-abigen-macro/tests/test_projects/enum_as_input/src/main.sw:12]: It's mostly because I believe that that's the conventional way to name your enum variants in Sway. I'm simply matching them against what the test expects, yes. 

The whole point of this PR is that what you write in Sway is what you get on the SDK side.

File: packages/fuels-core/src/code_gen/functions_gen.rs (hunk 4 above)
Comment (digorithm) [reply packages/fuels-core/src/code_gen/functions_gen.rs:744]: Oh no, that was by mistake (had it deleted while debugging some stuff), thanks for catching that!

File: packages/fuels-core/src/code_gen/custom_types_gen.rs (hunk 2 above)
Comment (digorithm) [reply packages/fuels-core/src/code_gen/custom_types_gen.rs:571]: Yup, naming these structs just as you would when writing Sway. I could keep it as `cocktail` but then the expected should then be `cocktail` and not `Cocktail`, as we're not mutating it anymore.

**Review by vnepveu — APPROVED** [review]
Good catch! LGTM

**Review by Salka1988 — DISMISSED** [review]

**Review by Br1ght0ne — DISMISSED** [review]

**Review by MujkicA — APPROVED** [review]

**Review by MujkicA — APPROVED** [review]
